posixlib/src/main/scala/scala/scalanative/posix/netdb.scala

```diff
@@ -1,21 +1,45 @@
 package scala.scalanative.posix
 
 import scalanative.unsafe._
+
 import scalanative.posix.sys.socket
 import scalanative.posix.netinet.in
 
+import scalanative.runtime.Platform
+
 @extern
 object netdb {
-  type addrinfo = CStruct8[CInt, CInt, CInt, CInt, socket.socklen_t, Ptr[
-    socket.sockaddr
-  ], Ptr[CChar], Ptr[Byte]]
+  /* This is the Linux layout. FreeBSD & macOS have the same size but
+   * swap ai_addr and ai_cannonname. FreeBSD documents this, macOS tells
+   * whoppers: it documents the Linux order in man pages and implements
+   * the FreeBSD layout.
+   *
+   * Handled in netdbOps below.
+   */
+
+  /* _Static_assert code in netdb.c checks that Scala Native and operating
+   * system structute definitions match "close enough". If you change
+   * something here, please make the corresponding changes there.
+   */
+
```

Comment:
These should probably be a formal Scaladoc - we do generate docs for these.  Int the C lib you can see them - https://javadoc.io/doc/org.scala-native/clib_native0.4_2.13/latest/scala/scalanative/libc/fenv$.html and also the next two. Complex has a comment on top.

The links to docs are on the main github page.

Reply:
Thank you for the timely review.

Eh?  Can you educate me on how it is created?  
Sorry but I do not know from Scaladoc.  

On a quick look, Scaladoc looks like Huge overhead for something that
is well described, modulo OS differences, on 'man' available on any system.

Camel can carry only so many bales.
I thought I was doing well writing it down at least once.

Reply:
I took a quick look at Scaladoc conventions. I am not adverse to someone doing
a PR after this one is merged to add Scaladoc but feel that lack of Scaladoc should
not hold up this PR.

Reply:
As just a note with no fancy stuff would suffice for what you have as a start. At the top of the `class`, `object` or just before any `type` or `def` you can put a comment.
```Scala
/** This is netdb that does xxx
 *
 *  Another paragraph saying how great it is
 */
object netdb { ...
```
Scalafmt will format it correctly.

Reply:
Trying to be a good kid, I checked the Contributor's Guide and noticed the total
absence of a section on Scaladoc and how to generate it within the Scala Native
build.   

Not to give in to a bad attitude, I figured out how to generate Scaladoc within
the Scala Native build. Old news to some, but not evident (sbt posixlib3/doc).
Posixlib was all that I needed, so I stopped before figuring out how to generate
the Scaladoc for all projects. 

Then I figured out how to view the local copy of the generated Scaladoc (location
and name of generated html files).  

What that showed me is a listing of objects, for each object there were methods and
types.  I checked this netdb.scala and a couple of others.  The descriptions
could certainly be improved by careful work, but they are, IMO, passable.

So, I believe that minimal Scaladoc comes for 'free' and ticks the
"is Scaladoc present" checkbox.  

Adding the items Eric suggests should be considered as desirable, but, again IMO, 
should not be a hard requirement.  Especially in the light of prior art and the total
lack of information in the Contributor's Guide.